### ISSUE TITLE
".des" no longer tried, when dataset extension not given

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/Ferret#1400]
**Reported by andrew wittenberg on 15 Jan 2014 21:27 UTC**
There appears to have been a recent change in behavior (perhaps as a result of the fix to #2087 ?) which is breaking many of our automated scripts.  Previously, when a dataset filename extension was not supplied, extensions (.des, .cdf, .nc) were tried in that order.  We took advantage of that behavior in scripts, so that they didn't need to know whether the dataset was a NetCDF file or a descriptor file.  But now it seems that the .des is no longer tried.  Could we return to the former behavior?

Old behavior:

```
 	NOAA/PMEL TMAP
 	FERRET v6.83 (beta)
 	Linux 2.6.32-279.5.1.el6.x86_64 64-bit - 08/22/12
 	15-Jan-14 16:21     

yes? sp touch a.des  ! dummy descriptor
yes? use a           ! ok, at least it tries...
 **TMAP ERR: Error in namelist record
             Didn't find: FORMAT_RECORD; or file not descriptor file
             Data set: ./a.des
yes? use b           ! correct response, for this non-existent file
 **TMAP ERR: non-existent or not on line
             Searched $FER_DESCR and $FER_DATA for b(.des,.cdf,.nc)
```

New behavior is different -- the .des extension isn't tried, and the error message doesn't even mention ".des" as one of the alternatives (and ".nc" is listed twice!):

```
 	NOAA/PMEL TMAP
 	FERRET v6.851 (beta)
 	Linux 2.6.32-431.el6.x86_64 64-bit - 12/12/13
 	15-Jan-14 16:20     

yes? sp touch a.des  ! dummy descriptor
yes? use a
 **TMAP ERR: non-existent or not on line
             Searched $FER_DESCR and $FER_DATA for a(.nc ,.cdf,.nc)
yes? use b
 **TMAP ERR: non-existent or not on line
             Searched $FER_DESCR and $FER_DATA for b(.nc ,.cdf,.nc)
```

Migrated-From: http://dunkel.pmel.noaa.gov/trac/ferret/ticket/2128

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 15 Jan 2014 22:37 UTC**
The default used to be .des, and some time back we changed that to netCDF, either .nc or .cdf.  I'll make sure we also try .des.

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 16 Jan 2014 00:01 UTC**
The change was the one in ticket #1531, making netcdf the default file type, and looking for .cdf or .nc. So this has been this way for a little while, since v6.84.

I've fixed it so that if no extension is given, Ferret looks in the dataset path directories for .cdf, .nc, and then .des. Here's what it looks like, including fixing the error message:

```
yes? sp ls duplicate* mc_test*
duplicate.cdf  duplicate.des  duplicate.nc  mc_test.des
 
! The one that's used is .cdf
use duplicate
sh dat
     currently SET data sets:
    1> ./duplicate.cdf  (default)
 name     title                     I      J      K      L      M      N
 SST      SEA SURFACE TEMPERATURE  1:1    1:10   ...    1:3    ...    ...
 
yes? use mc_test
yes? sh dat
     currently SET data sets:
    1> ./mc_test.des  (default)
 name     title                     I      J      K      L      M      N
 SST      SEA SURFACE TEMPERATURE  1:180  1:90   ...    1:3    ...    ...
 
yes? use nosuchfile
 **TMAP ERR: non-existent or not on line
             Searched $FER_DESCR and $FER_DATA for nosuchfile(.nc ,.cdf,.des)

```

I have not included .mc in these changes. In the fix for ticket #2087 we let .mc indicate a descriptor file, but we have never included ".mc" in this code which tries likely extensions.

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 16 Jan 2014 17:58 UTC**
Fixed in v6.86 (dated 1/15/14).
